include/openssl/crypto.h.in

```diff
@@ -90,6 +90,60 @@ int CRYPTO_atomic_or(uint64_t *val, uint64_t op, uint64_t *ret,
                      CRYPTO_RWLOCK *lock);
 int CRYPTO_atomic_load(uint64_t *val, uint64_t *ret, CRYPTO_RWLOCK *lock);
 
+# if defined(OPENSSL_SYS_WINDOWS) || defined(_WIN32) || defined(__CYGWIN__)
+typedef long CRYPTO_SIGNAL_type;
+# else
+typedef int  CRYPTO_SIGNAL_type;
+# endif
+
+# define CRYPTO_SIGNAL_DEFAULT_ACTION NULL
+
+typedef void (*CRYPTO_SIGNAL_ACTION)(CRYPTO_SIGNAL_type);
+
+typedef struct {
+    long signal;
+    CRYPTO_SIGNAL_ACTION action;
+} CRYPTO_SIGNAL;
+
+int CRYPTO_SIGNAL_block(CRYPTO_SIGNAL* p);
+int CRYPTO_SIGNAL_block_set(CRYPTO_SIGNAL** props);
+int CRYPTO_SIGNAL_unblock_all(void);
+
+typedef void *CRYPTO_MUTEX;
+typedef void *CRYPTO_CONDVAR;
```

Comment:
Again probably should be an anonymous structure. The `*` should not be part of the type.